ctru-sys/src/lib.rs

```diff
@@ -5,12 +5,10 @@
 #![allow(clippy::all)]
 
 pub mod result;
-
-mod bindings;
-
-pub use bindings::*;
 pub use result::*;
 
+include!(concat!(env!("OUT_DIR"), "/bindings.rs"));
```

Comment:
One reason I preferred a module previously was for `rust-analyzer`'s "go to definition", autocomplete, and doc comments to work a lot nicer (not sure if other IDEs / language servers have the same issue). It looks like this should be working now with https://github.com/rust-lang/rust-analyzer/issues/3767 closed, but when I check out your branch I can't seem to see any documentation, nor jump to these definitions.

Does your IDE work with the `include!()` like this? I wonder if I just have a misconfiguration — in any case, this probably isn't a big enough deal to block this change — but the developer experience is a lot nicer when IDE features work with it IMO.

Reply:
I just checked and I'm not getting code completion on the branch either, and the reason for that is because rust-analyzer is failing to build and run ctru-sys with these changes: https://gist.github.com/FenrirWolf/2944fb0378d2724f5ce3d6785a59276f

Looks like the culprit is that `TARGET` is set `x86_64-unknown-linux-gnu` instead of `armv6k-nintendo-3ds`, and so r-a passes `-m64` as a parameter and devkitARM doesn't like that.

Reply:
Ah, nice find! That gave me the idea to set `"rust-analyzer.cargo.target": "armv6k-nintendo-3ds"` in my config, and now it looks like everything is working the same as how it was when we were using a module!

I suppose this would be a good hint to add to our development wiki for people getting setup with `rust-analyzer` for general 3DS development.

Reply:
> Ah, nice find! That gave me the idea to set `"rust-analyzer.cargo.target": "armv6k-nintendo-3ds"` in my config, and now it looks like everything is working the same as how it was when we were using a module!
> 
> I suppose this would be a good hint to add to our development wiki for people getting setup with `rust-analyzer` for general 3DS development.

True, this is wiki material. Still, I prefer overriding the check command with `cargo-3ds` so that I can use incremental builds (damn you, env variables!).

```json
"rust-analyzer.check.overrideCommand": [
        "cargo",
        "3ds",
        "clippy",
        "--message-format=json"
    ],
```

Reply:
Yeah, I actually already had something like that configured, but it seems like for the build script to work correctly you _also_ need the `cargo.target` configured correctly. I'll see if I can add a section to the `Guides` page about this.